oracle-uri-proposal.md

```diff
@@ -0,0 +1,179 @@
+It can be used to mark the occurrence of some event (that may or may not ever happen).
+For example, it can be useful to mark the passage of time itself.
+
+- Example event id: `/time/2020-08-16T08:00:00?occur`
+- Example outcome sting:  `/time/2020-08-16T08:00:00?occur=true` (after the event has happened)
+
+### `weather`
+
+The weather has a number of discrete outcomes e.g. cloudy, raining, snowing, clear etc.
+
+- Example event id: `/Earth/Australia/Sydney?weather`
+- Example outcome id: `/Earth/Australia/Sydney?weather=☀️` (if there's sunny weather).
+
+### `digits(n_digits,least_significant_digit)`
+
+A positive number decomposed into its decimal digits.
+Each decimal digit is an event fragment i.e. the oracle will release a signature for each digit.
+
+It has two parameters:
+
+- `n_digits`: the number of digits to the left of the lowest digit.
+- `least_significant_digit`: (default=0) the position the lowest digit e.g. -1 for a number like `12.3` and 0 for `123`.
+
+If the actual value is above or equal to the maximum number expressible by the digits then the oracle must attest to the maximum number.
+
+For example, to attest to the bid price of Bitcoin on Binance between \$0 and \$99,999 the oracle would announce `/Binance/BTC_USDT/bid/2021-05-16T08:00:00?digits(5)`.
+If the bid price on the date is \$20,421 the oracle would release five signatures on the following messages:
+
+  1. `/Binance/BTC_USDT/bid-price/2021-05-16T08:00:00?digits(5).0=1`
+  2. `/Binance/BTC_USDT/bid-price/2021-05-16T08:00:00?digits(5).1=2`
+  3. `/Binance/BTC_USDT/bid-price/2021-05-16T08:00:00?digits(5).2=4`
+  4. `/Binance/BTC_USDT/bid-price/2021-05-16T08:00:00?digits(5).3=0`
+  5. `/Binance/BTC_USDT/bid-price/2021-05-16T08:00:00?digits(5).4=2`
+
+If `digits(5)` turns out to be insufficient later on because the bitcoin price appreciates too much the oracle can always announce a new `digits(6)` event on the same path.
+However, this new event would not share any of the same signatures/nonces; it would be a completely separate event.
+
+If a path ends in `<name1>_<name2>/bid-price/<datetime>` and has an event type of `digits` or `int` then the events value represents the bid price of `<name1>` in terms of `<name2>` at `<datetime>`.
+
+### `above(threshold)`
+
+This binary event type asserts that the quantity measured will be above a certain threshold.
+
+### `vs`
+
+This event type is for competitive matches where the one team wins and one team loses or there is a draw.
+Competitions where drawing is impossible can still use the `vs` event type and just use the `draw` outcome to indicate the match being cancelled due to some act of god.
+
+The `vs` event type requires that the path segment preceding it be in the form `<team1>_<team2>` where `<team1>_<team2>` are the competing teams.
+For example, `/NBA/s2019/2020-08-11/PHI_PHX?vs` describes a team named `PHI` is playing against another team named `PHX`.
+Additionally the client can infer the date of the match.
```

Comment:
@Tibo-lg made a point last meeting that you can have competitions with multiple entities. e.g. Top scorer in a competition. For this you could do `/FIFA-world-cup/2021/top-scoring-player/Messi_Mbappe_Ronaldo?vs(3)`.